src/frontend/components/property-type/base-property-component.tsx

```diff
@@ -108,6 +111,17 @@ const BasePropertyComponent: React.FC<BasePropertyComponentProps> = (props) => {
     )
   }
 
+  if (baseProperty.type === 'key-value') {
```

Comment:
Would not be safer to use `enum` here? Generally to refactor `PropertyType` from `type` to `enum`?